MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -1094,6 +1116,13 @@
         }
     }
 
+#if MVK_MACOS
+    if (mvkOSVersionIsAtLeast(10.14)) {
+        static const uint32_t kAMDVendorId = 0x1002;
+        _metalFeatures.subgroupSize = (_properties.vendorID == kAMDVendorId) ? 64 : 32;
```

Comment:
This feels a bit hacky and subject to future unexpected behaviour. 

Is creating a `MTLComputePipelineState` and using `threadExecutionWidth` the only way to get this programmatically? 

Or is it possible that this test could be done with reference to `MTLFeatureSet` instead (AMD referencing a particular GPU type)?

Reply:
> Is creating a `MTLComputePipelineState` and using `threadExecutionWidth` the only way to get this value programmatically?

Currently, yes. I've asked them to provide queries for the minimum and maximum thread execution width, but I think I asked for it too late for macOS 11.

Reply:
> Currently, yes

And 64/32 it can't realistically be differentiated via a `MTLFeatureSet` (ie. GPU family)?

Reply:
> > Currently, yes
> 
> And 64/32 it can't realistically be differentiated via a `MTLFeatureSet` (ie. GPU family)?

No. Modern AMD and Intel chips both support `MTLGPUFamilyMac2`, but Intel uses 32 while AMD uses 64.